### PR TITLE
feature/ICL-120 guides

### PR DESCRIPTION
**Pull request by JGiter**
https://energyweb.atlassian.net/browse/ICL-120

### CODE REVIEW

File: docs/guides/application.md [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+    "owner": "0x5f7***",
+    "namespace": "kyc.apps.OKE.iam.ewc",
+    "namehash": "0xad98bf932b0b96385c3d28a0b343d3f94a3442a63b170df31815a1245163089e",
+    "definition": {
+        "appName": "IAM KYC",
+        "logoUrl": "",
+        "websiteUrl": "https://www.energyweb.org/",
+        "description": "Temp KYC app"
+    }
+}
```
Comment (jrhender) [inline docs/guides/application.md:22]: Better to use the more official/professional EnergyWeb app
```suggestion
{
  "id": 347,
  "name": "verification",
  "owner": "0x627306090abaB3A6e1400e9345bC60c78a8BEf57",
  "namespace": "verification.apps.energyweb.iam.ewc",
  "namehash": "0x39f219f48798d619edc9d5ab0f5f802a86383b57a99965722a4cb9d83d3fd16a",
  "definition": {
    "appName": "IAM KYC App",
    "logoUrl": "",
    "websiteUrl": "https://www.energyweb.org/",
    "description": ""
  }
}
```

File: docs/guides/organization.md [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
-/*
-TO DO: 
-//Define Organization in context of iam-client-lib/Switchboard
-//Define Suborganization in context of iam-client-lib/Switchboard
+[Organization Interface](../api/interfaces/cacheServerClient_cacheServerClient_types.IOrganization.md)
 
-//Give an example(s) of an organization
-//Give an example(s) of a sub-organization
-*/
+**Organization** is basic type in EnergyWeb domains tree. Its main purpose is to be a host.
```
Comment (jrhender) [inline docs/guides/organization.md:5]: What do you mean by "be a host"?

File: docs/guides/organization.md [hunk 3]
```diff
@@ -1,17 +1,37 @@
...
+All other domains such as applications and roles should have parent organization.
+If needed organization can include other sub-organizations.Organization is represented by
```
Comment (jrhender) [inline docs/guides/organization.md:7]:
```suggestion
If needed organization can include other sub-organizations. Organization is represented by
```

File: docs/guides/organization.md [hunk 4]
```diff
@@ -1,17 +1,37 @@
...
+[`IOrganization`](../api/interfaces/IOrganization.md) interface. Organization can
+be created either as way to structure system or group correlated entities along with permissions hieararchy.
```
[the remark this answers is not in the document]
Comment (jrhender) [reply docs/guides/organization.md:9]: What do you mean by "structure system" here?

File: docs/guides/organization.md [hunk 5]
```diff
@@ -1,17 +1,37 @@
...
-//Define Suborganization in context of iam-client-lib/Switchboard
+[Organization Interface](../api/interfaces/cacheServerClient_cacheServerClient_types.IOrganization.md)
 
-//Give an example(s) of an organization
-//Give an example(s) of a sub-organization
-*/
+**Organization** is basic type in EnergyWeb domains tree. Its main purpose is to be a host.
+All other domains such as applications and roles should have parent organization.
+If needed organization can include other sub-organizations.Organization is represented by
+[`IOrganization`](../api/interfaces/IOrganization.md) interface. Organization can
```
Comment (jrhender) [inline docs/guides/organization.md:8]:
```suggestion
[`IOrganization`](../api/interfaces/IOrganization.md) interface. Organizations can
```

File: docs/guides/organization.md [hunk 6]
```diff
@@ -1,17 +1,37 @@
...
+    "orgName": "OKE"
+  },
+  "apps": [
+    ...
+  ],
+  "subOrgs": [],
+  "roles": [
+    ...
+  ]
+}
```
Comment (jrhender) [inline docs/guides/organization.md:30]: Better to use the EnergyWeb org here as it is more professional
```suggestion
{
  "id": 24,
  "name": "energyweb",
  "namespace": "energyweb.iam.ewc",
  "namehash": "0x9c0d998fec800b14f2aa5b83643561cab9a7f52630b5c3889cf5e2a7a175d632",
  "owner": "0x627306090abaB3A6e1400e9345bC60c78a8BEf57",
  "definition": {
    "others": {
      "logoUrl": "https://stake.energyweb.org/assets/img/ew-flex-single-logo.png"
    },
    "logoUrl": "https://stake.energyweb.org/assets/img/ew-flex-single-logo.png",
    "orgName": "Energy Web Foundation",
    "websiteUrl": "https://energyweb.org",
    "description": "Energy Web (EW) is a global nonprofit organization accelerating a low-carbon, customer-centric electricity system by unleashing the potential of open-source, decentralized technologies. EW focuses on building core infrastructure and shared technology, speeding adoption of commercial solutions, and fostering a community of practice."
  },
  "subOrgs": []
}
```

File: docs/guides/role.md [hunk 7]
```diff
@@ -1,19 +1,54 @@
...
 
-/*
-TO DO: 
-//Define Role in context of iam-client-lib/Switchboard
+[Role Interface](../api/interfaces/cacheServerClient_cacheServerClient_types.IRole.md)
 
-//Give an example(s) of a Role
-*/
+**Role** is leaf in EnergyWeb domains tree. Its main purpose is to distribute responsibilities
+and permissions in some domain. As part of its definition, role determines the order of its
```
Comment (jrhender) [inline docs/guides/role.md:6]: I don't think "order" is the right word here
```suggestion
and permissions in some domain. As part of its definition, role determines the permissions of its
```

File: docs/guides/role.md [hunk 8]
```diff
@@ -1,19 +1,54 @@
...
+issuance and revoсation. In particular requesting of role can require preliminary assignment
+of some other role. The similar applies to issuance. Role can be placed under organization or
+application thus defining its scope. Organization role serves organization government purposes,
+while application roles mainly describes application clients.
```
Comment (jrhender) [inline docs/guides/role.md:10]: I would remove this bit. I'm not sure what "government purposes" refers to here.
```suggestion
application thus defining its scope.
```

**Review by jrhender — CHANGES_REQUESTED** [review]
@JGiter Thanks for moving this ahead. Some suggestions to look at when you have a moment

File: docs/guides/organization.md (hunk 2 above)
Comment (JGiter) [reply docs/guides/organization.md:5]: I tried to express that organization serves as container or root which other domains attached to. May be there is more appropriate term for this?

Comment (jrhender) [reply docs/guides/organization.md:5]: I think that either of the terms, "container" or "root", would be good, however how about the text below?
```suggestion
**Organization** is basic type in EnergyWeb domains tree. Its main purpose is to be a root that can group other domains, such as apps and roles.
```

**Review by jrhender — APPROVED** [review]
One other comment, but looks good from my end.

**Comment by jrhender** [comment]
@JGiter Can we move this PR to "review for review" and merge? Why is it still in draft?

**Pull request merged**